src/serlo-editor-repo/plugin-text/components/link/link-content-search.tsx

```diff
@@ -0,0 +1,283 @@
+    })
+
+    return () => {
+      timeout && clearTimeout(timeout)
+    }
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [value])
+
+  useEffect(() => {
+    const effectFetch = async () => {
+      const fetchedData = await fetchQuickbarData()
+      if (fetchedData) setData(fetchedData)
+    }
+
+    if (query && !data) {
+      // eslint-disable-next-line no-console
+      effectFetch().catch(console.error)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [query, data])
+
+  useEffect(() => {
+    setSel(0)
+  }, [query, data, value])
+
+  const results = data ? findResults(data, query) : []
+
+  function setHref(index?: number) {
+    const activeIndex = index ?? sel
+    if (activeIndex < 0 || activeIndex > results.length) return
+
+    const href =
+      activeIndex < results.length ? `/${results[activeIndex].entry.id}` : query
+
+    if (
+      checkSerloHref(href) ||
+      href.startsWith('http://') ||
+      href.startsWith('https://')
+    ) {
+      setValue(href)
+    } else {
+      showToastNotice(
+        'Hmm, das schaut leider nach einem ungültigen Link aus.',
+        'warning'
+      )
+    }
+  }
+
+  const onKeyDown = (e: KeyboardEvent<HTMLInputElement>) => {
+    {
+      if (e.key === 'Escape') {
+        if (query) setQuery('')
+        else removeLink()
+      }
+      if (e.key === 'ArrowDown' || e.key === 'ArrowUp' || e.key === 'Enter') {
+        if (e.key === 'ArrowDown' && sel < results.length) {
+          setSel(sel + 1)
+        }
+        if (e.key === 'ArrowUp' && sel >= 0) {
+          setSel(sel - 1)
+        }
+        if (e.key === 'Enter') {
+          setHref()
+        }
+        e.preventDefault()
+        return false
+      }
+    }
+  }
+
+  return (
+    <>
+      {isEditMode ? (
+        <>
+          <label className="block px-side pt-4">
+            Suche einen Serlo Inhalt oder füge einen Link ein …
+          </label>
+          <div className="relative w-[27rem]">
+            {renderInput()}
+            {query && renderResetButton()}
+          </div>
+          {renderResults()} <br />
+        </>
+      ) : (
+        <>{renderSetEntry()}</>
+      )}
+    </>
+  )
+
+  function renderInput() {
+    return (
+      <input
+        type="text"
+        className="mx-side w-[25rem] border-3 border-editor-primary-50 bg-editor-primary-100 rounded-xl mt-2 pl-3 pr-12 h-12 align-end hover:border-editor-primary-100 focus:border-editor-primary-200 outline-none"
+        value={query}
+        onChange={(value) => setQuery(value.target.value)}
+        placeholder="Stichwort oder Link"
+        ref={inputRef}
+        onKeyDown={onKeyDown}
+      />
+    )
+  }
+  function renderResetButton() {
+    return (
+      <div
+        className={clsx(
+          'absolute top-4 right-6 flex items-center justify-center w-8 h-8',
+          'serlo-button-editor-secondary',
+          'cursor-pointer text-gray-700'
+        )}
```

Comment:
Just out of curiosity, what is the need for using `clsx` here?

Reply:
well we talked about keeping the tailwind lines from becoming super long and `clsx` is an easy way to do that.
But tbh. this is more a leftover when I had more rules in this className.

Reply:
(and at least I'm not at all consistent with that, some lines above is a super long className…)